code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -331,9 +331,10 @@ public void setMapDao(GenericMapDao mapDao) {
                         .on(INSTANCE_HOST_MAP.INSTANCE_ID.eq(INSTANCE.ID))
                     .where(INSTANCE_HOST_MAP.HOST_ID.eq(hostId)
                         .and(INSTANCE.REMOVED.isNull())
+                        .and(INSTANCE.ID.ne(instanceId))
                         .and(INSTANCE.STATE.in(InstanceConstants.STATE_STARTING, InstanceConstants.STATE_RESTARTING, InstanceConstants.STATE_RUNNING))
                         .and(INSTANCE_HOST_MAP.REMOVED.isNull())
-                        .and(PORT.STATE.in(CommonStatesConstants.ACTIVE, CommonStatesConstants.ACTIVATING))
+                        .and(PORT.STATE.notIn(CommonStatesConstants.REMOVED, CommonStatesConstants.REMOVING, CommonStatesConstants.PURGED, CommonStatesConstants.PURGING))
```

Comment:
Instead of listing all these states I think PORT.REMOVED.isNull() would be sufficient